core/src/main/java/org/apache/gravitino/connector/PropertiesMetadata.java

```diff
@@ -149,11 +149,13 @@ default PropertyEntry<?> getPropertyEntry(String propertyName) throws IllegalArg
     if (!containsProperty(propertyName)) {
       throw new IllegalArgumentException("Property is not defined: " + propertyName);
     }
+
     return getNonPrefixEntry(propertyName).isPresent()
         ? getNonPrefixEntry(propertyName).get()
-        : getPropertyPrefixEntry(propertyName).get();
+        : (getPropertyPrefixEntry(propertyName).isPresent()
+            ? getPropertyPrefixEntry(propertyName).get()
+            : null);
```

Comment:
This appears somewhat complex at first glance, so it may be better to use an if statement. I'm also concerned that returning null could cause a NPE elsewhere, so it may be better to throw an exception. You might want to double-check the unit tests for this as well.